plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -902,11 +902,45 @@ func PodFitsHostPorts(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 	}
 
 	existingPorts := nodeInfo.UsedPorts()
-	for wport := range wantPorts {
-		if wport != 0 && existingPorts[wport] {
-			return false, []algorithm.PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
+
+	// try to see whether two hostPorts will conflict or not
+	for existingPort := range existingPorts {
```

Comment:
As we slack, the logic is correct. Can we move those check to a method/function, and change it to the following format?

```
for wantPort := range wantPorts  {
  if portConflict(wantPort, existedPorts) {
    return
  }
}
```

And describe the detail in `portConflict` comments.

Reply:
Make sense, changed according to your suggestion.

Reply:
Can we just handle it in one loop, and put more logic in portConflict?

Reply:
I once had the same thought as you, try to use one loop to finish the job. But I can't find it in the end. Because it is indeed a double check as we talked on slack. I suggest using another map which can reduce it to one for loop, but you said it is not worth it.I think for now it meets your "more readable " need.

Reply:
I will try to put more in portConflict